lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -18,17 +18,23 @@ class MissingUniqueIndexes < Base # :nodoc:
 
       private
 
-      def message(table:, columns:)
-        # rubocop:disable Layout/LineLength
-        "add a unique index on #{table}(#{columns.join(', ')}) - validating uniqueness in the model without an index can lead to duplicates"
-        # rubocop:enable Layout/LineLength
+      # rubocop:disable Layout/LineLength
+      def message(model:, table:, columns:, problem:)
+        case problem
+        when :validations
+          "add a unique index on #{table}(#{columns.join(', ')}) - validating uniqueness in the model without an index can lead to duplicates"
+        when :has_ones
+          "add a unique index on #{table}(#{columns.first}) - using `has_one` in the #{model.name} model without an index can lead to duplicates"
```

Comment:
I suggest we use `columns.join(', ')` - the result will be the same for single-entry arrays but we'll immediately see something is wrong in case we pass a larger array in a future version.